src/ZkFold/Base/Protocol/ARK/Plonk/Relation.hs

```diff
@@ -38,11 +38,14 @@ toPlonkRelation :: forall n l a .
     => Arithmetic a
     => Scale a a
     => Vector l Natural
-    -> ArithmeticCircuit a Par1
+    -> ArithmeticCircuit a (Vector l) Par1
     -> Maybe (PlonkRelation n l a)
 toPlonkRelation xPub ac0 =
     let ac = desugarRanges ac0
-        evalX0 = evalPolynomial evalMonomial (\x -> if x == 0 then one else var x)
+
+        varF (NewVar ix) = if ix == 0 then one else var (ix + value @l)
```

Comment:
So after this PR we need to decide which input is the constant 1, and `NewVar 0` looks like a good choice.

Reply:
Yes, after we optimize constants we won't need a special variable for `one` or any other constant.

Reply:
I remember @vlasin saying that this variable is still needed in zk proving algorithms somehow, or do I remember it wrong?

Reply:
I remember the opposite 🤷

Reply:
No, it's not needed. However, some minor changes will be required in the `toPlonkRelation` function.

Reply:
@echatav, if all tests pass, you can resolve this conversation

Reply:
@TurtlePU Neither my laptop nor CI are adequate to check if all tests pass :-/

Reply:
@echatav it's ok, I'll run them myself then once merge conflicts are resolved

Reply:
I just want to point out that there is a very unlikely scenario when the quoted line results in "index overflow". This is something we want to solve in the var generation PR.

Reply:
Sorry, my bad! We changed this conversion algorithm some time ago, so it should not cause any problems.

Reply:
I'd like to have a discussion to understand var generation better.